### PR TITLE
Adding a tags filter for collection versions

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #5571
https://pulp.plan.io/issues/5571

### CODE REVIEW

File: pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        collections = self.client.get(ANSIBLE_COLLECTION_VERSION_CONTENT_PATH, params=params)
+        self.assertEqual(len(collections), 0, collections)
+
+        params = {"tags": "k8s"}
+        collections = self.client.get(ANSIBLE_COLLECTION_VERSION_CONTENT_PATH, params=params)
+        self.assertEqual(len(collections), 2, collections)
+
+        params = {"tags": "k8s,kubernetes"}
+        collections = self.client.get(ANSIBLE_COLLECTION_VERSION_CONTENT_PATH, params=params)
+        self.assertEqual(len(collections), 2, collections)
```
Comment (fabricio-aguiar) [inline pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py]: cool!

File: pulp_ansible/app/viewsets.py [hunk 2]
```diff
@@ -129,9 +134,25 @@ def filter_by_q(self, queryset, name, value):
...
+            qs (django.db.models.query.QuerySet): CollectionVersion queryset
+            value (string): A comma separated list of tags
+
+        Returns:
+            Queryset of CollectionVersion that matches all tags
+
+        """
+        for tag in value.split(","):
+            qs = qs.filter(tags__name=tag)
+        return qs
```
Comment (fabricio-aguiar) [inline pulp_ansible/app/viewsets.py]: I think we can get rid of the loop, and do only one query, changing to:
```python
qs = qs.filter(tags__name__in=value.split(","))
```

File: pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        body = gen_ansible_remote(url=ANSIBLE_COLLECTION_TESTING_URL)
+        remote = self.client.post(ANSIBLE_COLLECTION_REMOTE_PATH, body)
+        self.addCleanup(self.client.delete, remote["pulp_href"])
+
+        # Sync the repository.
+        sync(self.cfg, remote, repo)
+
+        # filter collection versions by tags
+        params = {"tags": "nada"}
```
Comment (kersommoura) [inline pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py]: Nice!

File: pulp_ansible/app/viewsets.py (hunk 2 above)
Comment (daviddavis) [reply pulp_ansible/app/viewsets.py]: They have slightly different behavior:

```
In [1]: CollectionVersion.objects.filter(tags__name="nada").filter(tags__name="k8s")                                                         
Out[1]: <QuerySet []>                                                                                                                       

In [2]: CollectionVersion.objects.filter(tags__name__in=["nada", "k8s"])
Out[2]: <QuerySet [<CollectionVersion: k8s>, <CollectionVersion: k8s_demo_collection>]>
```

We want the first behavior. I will add a test for this.

File: pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        collections = self.client.get(ANSIBLE_COLLECTION_VERSION_CONTENT_PATH, params=params)
+        self.assertEqual(len(collections), 0, collections)
+
+        params = {"tags": "k8s"}
```
Comment (kersommoura) [inline pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py]: Just a suggestion. If those collections will be reused, or modified perhaps we can move them to a constant.

File: pulp_ansible/app/viewsets.py (hunk 2 above)
Comment (fabricio-aguiar) [reply pulp_ansible/app/viewsets.py]: Cool! Like `or` and `and`

File: pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py (hunk 4 above)
Comment (daviddavis) [reply pulp_ansible/tests/functional/api/collection/test_crud_collection_versions.py]: I don't know we will reuse the list of tags but will keep in mind if we ever do.

**Pull request merged**